src/fileio/fileio_scen.cpp

```diff
@@ -1066,6 +1070,63 @@ void readScenarioFromXml(ticpp::Document&& data, cScenario& scenario) {
 		throw xMissingElem("scenario", *reqs.begin(), data.FirstChildElement()->Row(), data.FirstChildElement()->Column(), fname);
 }
 
+void readEditorStateFromXml(ticpp::Document&& data, cScenario& scenario) {
+	using namespace ticpp;
+	int maj, min, rev;
+	std::string fname, type, name, val;
+	editor_state_t& editor_state = scenario.editor_state;
+	initialXmlRead(data, "editor", maj, min, rev, fname);
+	Iterator<Attribute> attr;
+	Iterator<Element> elem;
+
+	std::string child_type;
+	short num;
+	location section;
+	location center;
+	short viewing_mode;
+
+	for(elem = elem.begin(data.FirstChildElement()); elem != elem.end(); elem++) {
+		elem->GetValue(&type);
+		if(type == "last-out-section") {
+			editor_state.last_out_edited = readLocFromXml(*elem);
+		}else if(type == "last-town") {
+			elem->GetText(&editor_state.last_town_edited);
+		}else if(type == "drawing"){
+			editor_state.drawing = str_to_bool(elem->GetText());
+		}else if(type == "editing-town"){
+			editor_state.editing_town = str_to_bool(elem->GetText());
+		}else if(type == "town-view-state"){
+			Element* child = elem->FirstChildElement();
+			while(child != nullptr){
```

Comment:
Please use an `Iterator<Element>` for this.

Reply:
When I set it up that way,

```
Iterator<Element> child;              
for(child = child.begin(elem->FirstChildElement()); child != child.end(); child++){
```

The loop body never executes. I swear I'm following the exact same pattern as the enclosing loop, but it doesn't work, begin() returns end() even when elem->FirstChildElement() returns the proper result. When I switched to the pointer way it did work, and I just did not care to debug a non-trivial tinyxml bug when I can't find documentation of how the iterator is supposed to work.

Reply:
Ohhhh. I think I see the problem, and tutorial_ticpp.txt actually does have info on this. Another thing I'd like to make easier for new contributors to find and learn from.

Reply:
It has to be like this:

```
			for(child = child.begin(&(*elem)); child != child.end(); child++){
```
Dereferencing the iterator to get the Element, then passing its address to child.begin(). What misled me was that the enclosing loop is calling FirstChildElement() on the document, but when doing that on the inner loop it actually tries to iterate the inner child. This all makes sense to me now.

Reply:
or instead of `&(*`, `.Get()`